pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
The object is still available, just within the context of the call that passes in the old object.

Off the top of my head, something like this:

```
func (s *Storage) Update(ctx api.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error) {
	if user, ok := api.UserFrom(ctx); ok {
		if s.superUser != "" && user.GetName() == s.superUser {
			return s.StandardStorage.Update(ctx, name, objInfo)
		}
	}
	
	wrappedInfo := &clusterRoleEscalationCheckingUpdatedObjectInfo{objInfo, s.ruleResolver}

  	return s.StandardStorage.Update(ctx, name, wrappedInfo)
}

type clusterRoleEscalationCheckingUpdatedObjectInfo struct {
  info rest.UpdatedObjectInfo
  ruleResolver validation.AuthorizationRuleResolver
}

func (n *clusterRoleEscalationCheckingUpdatedObjectInfo) Preconditions() *api.Preconditions {
  return n.info.Preconditions()
}

func (n *clusterRoleEscalationCheckingUpdatedObjectInfo) UpdatedObject(ctx api.Context, oldObj runtime.Object) (runtime.Object, error) {
	obj, err := n.info.UpdatedObject(ctx, oldObj)
	if err != nil {
		return obj, err
	}

	clusterRoleBinding := obj.(*rbac.ClusterRoleBinding)
	rules, err := s.ruleResolver.GetRoleReferenceRules(ctx, clusterRoleBinding.RoleRef, clusterRoleBinding.Namespace)
	if err != nil {
		return nil, err
	}
	if err := validation.ConfirmNoEscalation(ctx, n.ruleResolver, rules); err != nil {
		return nil, err
	}
	return obj, nil
}
```